d2-schemas/src/main/pegasus/com/linkedin/d2/D2RingProperties.pdl

```diff
@@ -0,0 +1,92 @@
+namespace com.linkedin.d2
+
+/**
+ * D2 hash ring properties used for load balancing.
+ */
+record D2RingProperties {
+
+  /**
+   * How fast should load balancer ramp up node that has been 100% degraded and is being recovered. Defaults to 1.0.
+   */
+  ringRampFactor: optional double
+
+  /**
+   * The number of points per weight given for a particular node in a hash ring. Defaults to 100.
+   */
+  pointsPerWeight: optional int
+
+  /**
+   * What hashing method used to hash incoming request. Used to determine which node should serve the request. Choices are Random, UriRegex
+   */
+  @symbolDocs = {
+    "URI_REGEX" : "Extract a key from URI and use it to hash",
+    "RANDOM" : "The default fall back value"
+  }
+  hashMethod: optional enum HashMethod {
+    RANDOM
+    URI_REGEX
+  }
+
+  /**
+   * Configuration used to supplement the hash method.
+   */
+  hashConfig: optional record HashConfig {
+
+    /**
+     * List of Regex to match against the URI of incoming request and compute hash value. The hash value is computed based on the contents of the first capture group of the first expression that matches the request URI
+     */
+    uriRegexes: optional array[string]
+
+    /**
+     * Optional config value. if true, fail if no regex matches, otherwise fall back to random.
+     */
+    failOnNoMatch: boolean = false
+
+    /**
+     *  Optional config value. if false, don't warn on falling back to random if the uri doesn't match the regex
+     */
+    warnOnNoMatch: boolean = true
+  }
```

Comment:
I thought relative LB does not support affinity routing? If that's the case you do not need these fields.

Reply:
It is supported. Underlying ring behavior is unchanged. Relative LB is for adjusting health of servers (points on ring).

Reply:
OK I misunderstood that part.